handlers/grafana.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"k8s.io/api/core/v1"
+
+	"github.com/swift-sunshine/swscore/config"
+	"github.com/swift-sunshine/swscore/kubernetes"
+	"github.com/swift-sunshine/swscore/log"
+	"github.com/swift-sunshine/swscore/models"
+)
+
+// GetGrafanaInfo provides the Grafana URL and other info, first by checking if a config exists
+// then (if not) by inspecting the Kubernetes Grafana service in namespace istio-system
+func GetGrafanaInfo(w http.ResponseWriter, r *http.Request) {
+	info, code, err := getGrafanaInfo(getGrafanaService)
+	if err != nil {
+		log.Error(err)
+		RespondWithError(w, code, err.Error())
+		return
+	}
+	RespondWithJSON(w, code, info)
+}
+
+func getGrafanaService() (*v1.ServiceSpec, error) {
+	client, err := kubernetes.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	details, err := client.GetServiceDetails("istio-system", "grafana")
```

Comment:
I would be careful with hardcoding the assumption that we always reside in namespace istio-system

Reply:
It's not especially "we", Grafana can be deployed there when using the Istio setup for kubernetes, without SWS involved. Afaik it's the standard configuration.
But it's right that I can make it more flexible and make these names configurable.

Also note that the whole Grafana URL can be overridden in config so the users would not be stuck in using Istio default deployment.